### PR TITLE
unisonlang M5c

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by cho-m** [comment]
I think the test needs to run `project.create` in CLI but there doesn't seem to be an easy way to do it.

Locally tried to do it through PTY but haven't had any success. Passing in command after timeout results in input getting chopped like `ect.create`. Using `IO#expect` to wait on prompt just gets stuck. Did try common settings like using xterm and modifying winsize.

**Comment by cho-m** [comment]
Linux needs `ncurses`. Doesn't seem to be used on macOS.

System error on 13-arm64 and 12-arm64 makes me think some part of build is trying to use an x86_64 binary.
```
  ERROR in ./src/UnisonLocal.elm
  Module build failed (from ./node_modules/elm-webpack-loader/index.js):
  Compiler process exited with error Error attempting to run Elm compiler "elm":
  Error: spawn Unknown system error -86
   @ ./src/unisonLocal.js 15:0-40 45:0-20
  
  webpack 5.66.0 compiled with 1 error in 7214 ms
```

File: Formula/u/unisonlang.rb [hunk 1]
```diff
@@ -24,18 +24,26 @@ class Unisonlang < Formula
...
   depends_on "haskell-stack" => :build
   depends_on "node@18" => :build
 
   uses_from_macos "python" => :build
   uses_from_macos "xz" => :build
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "ncurses"
+  end
```
Comment (cho-m) [inline Formula/u/unisonlang.rb:37]: macOS:
```
ucm:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1775.118.101)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1122.33.0)
	/usr/lib/libcharset.1.dylib (compatibility version 2.0.0, current version 2.0.0)
```

Linux:
```
  NEEDED       libncursesw.so.6
  NEEDED       libz.so.1
  NEEDED       libc.so.6
  NEEDED       libm.so.6
```

**Comment by chenrui333** [comment]
:LGTM:

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**